drivers/i2c/i2c_mcux_flexcomm.c

```diff
@@ -193,23 +203,50 @@ static int mcux_flexcomm_transfer(const struct device *dev,
 }
 
 #if defined(CONFIG_I2C_TARGET)
+
+static int mcux_flexcomm_target_find_free_target(struct mcux_flexcomm_data *data)
+{
+	for (int i = 0; i < MCUX_FLEXCOMM_MAX_SLAVES; i++) {
+		if (!data->targets[i].target_attached) {
+			return i;
+		}
+	}
+	return -EBUSY;
+}
+
+static int mcux_flexcomm_target_find_target_by_addr(struct mcux_flexcomm_data *data,
+					     uint16_t addr)
+{
+	for (int i = 0; i < MCUX_FLEXCOMM_MAX_SLAVES; i++) {
+		if (data->targets[i].target_cfg->address == addr) {
```

Comment:
this should probably also check target_attached to ensure we don't get a stale entry (E.G. you don't clear address when freeing).

I think the change would be easier to read if you split it up in a small series, E.G.:
- add struct mcux_flexcomm_target_config and update code to use target pointer
- add (dummy) _find_free_target / _find_target_by_addr
- support up to 4 slave addresses

Reply:
Reworked to split into seperate commits as suggested.
Also added proper unregistering and cleanup.